main.go

```diff
@@ -0,0 +1,215 @@
+	db, err := sql.Open("postgres",
+		fmt.Sprintf("user=%s password=%s dbname=%s host=%s port=%s sslmode=%s",
+			PG_USER,
+			PG_PASSWORD,
+			PG_DB,
+			PG_HOST,
+			PG_PORT,
+			PG_SSL,
+		))
+	if err != nil {
+		log.Fatalf("Error connecting to Postgres => %s", err.Error())
+	}
+	log.Printf("PQ Database connection made to %s", PG_DB)
+	return db
+}
+
+// handleFile processes new files
+//
+// The file is read into memory, parsed then inserted to the database.
+func handleFile(filename string, db *sql.DB) {
+	log.Printf("Processing, %s", filename)
+	fileContents, err := ioutil.ReadFile(filename)
+	if err != nil {
+		log.Printf("Failed to read in file, %s => %s", filename, err.Error())
+		return
+	}
+
+	tm, err := getDate(filename)
+	if err != nil {
+		log.Printf("Failed to parse date from file, %s, ignored.", filename)
+		return
+	}
+
+	data, err := parseData(tm, fileContents)
+	if err != nil {
+		log.Printf("Failed to parse data from %s => %s", filename, err.Error())
+		return
+	}
+
+	if err = dataset(data).insert(db); err != nil {
+		log.Printf("Failed to insert data from, %s => %s", filename, err.Error())
+	}
+}
+
+// Update the materialized views listed in `materializedViews`
+func updateViews(db *sql.DB) {
+	txn, err := db.Begin()
+	if err != nil {
+		log.Printf("ERR: failed to start pq txn for materialized view updates => %s", err.Error())
+		return
+	}
+
+	for _, view := range materializedViews {
+		if _, err = txn.Exec(fmt.Sprintf("REFRESH MATERIALIZED VIEW %s", view)); err != nil {
+			log.Printf("Failed to update materialized view, %s => %s", view, err.Error())
+		}
+	}
+
+	if nil == txn.Commit() {
+		log.Println("materialized views updated")
+	}
+}
+
+func LoadAllFiles(watchDir string) {
+	log.Printf("Loading all files in directory, %s", watchDir)
+
+	db := dbConnect()
+	defer db.Close()
+
+	files, err := ioutil.ReadDir(watchDir)
+	if err != nil {
+		log.Fatalf("Failed to read in directory info => %s", err.Error())
+	}
+
+	// handle every data file
+	for _, f := range files {
+		handleFile(path.Join(watchDir, f.Name()), db)
+	}
+
+	updateViews(db) // refresh the materialized views afterwards
+}
+
+func watchDirectory(watchDir string) {
+	// start watching for new files
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.Fatal("Failed to instantiate file watcher")
+	}
+	defer watcher.Close()
+
+	// start the file system watcher
+	if err = watcher.WatchFlags(watchDir, fsnotify.FSN_CREATE); err != nil {
+		log.Fatalf("Failed to start watching directory, %s => %s", watchDir, err.Error())
+	}
+
+	// wait for any new files to be added, then process them
+	for {
+		select {
+		case event := <-watcher.Event:
+			// reconnect to the DB for each event because otherwise the connection gets stale
+			db := dbConnect()
+			if filenameRegex.MatchString(event.Name) {
+				// sleep to allow the whole file to be transmitted.
+				// otherwise we get a parsing error because it's incomplete.
+				time.Sleep(time.Duration(2 * time.Second))
```

Comment:
Is there a better way to do this other than sleeping?  What if the file is massive or something?  I guess 2s is fine but stil.